text/0037-minting.md

```diff
@@ -0,0 +1,167 @@
+
+### Multiple Asset Fees
+
+The startup-config for the consensus validator node contains the governors, as well as a `TokensConfig`, which specifies the following per token:
+
+* Token ID
+* Minimum fee
+* Whether any fee is allowed 
+* The list of governors who can change the Minting Config for this token (issue `MintConfigTx` transactions)
+
+Every transaction's fee is paid in the same token type as the inputs and outputs of the transaction.
+
+Prioritization on the consensus validator nodes is determined from the fee of the transaction, multiplied by a priority number derived from the minimum fee for the token asset (see the [Tx fee priority commit](https://github.com/mobilecoinfoundation/mobilecoin/commit/9450b1966cef23cedd53cac5f169110019fe2f2d). Therefore, prioritization can occur without revealing the token types involved in the transactions in a block.
+
+A client can adjust its fee during times of congestion by knowing both the minimum fee for the token type they are attempting to transact, along with the rejected fee value they submitted.
+
+Currently, the enclaves aggregate the fee for multiple transactions in the same block so that there is only one fee output in order to save space in the ledger. With multiple fees, the aggregation will produce a number of fee outputs capped by the minimum of either the number of token IDs in the block or the number of transactions in a block.
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Adding multiple confidential asset types with minting, burning, and bridge operation increases the complexity of the system.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+## Differences with USDC contract design
+
+The [USDC smart contract](https://github.com/centrehq/centre-tokens/blob/master/doc/tokendesign.md) similarly has, a lowest level `minter` role, corresponding roughly to our `signers`, and `master minters` which increase the mint allowance of individual minters, similarly to our `governor` role. The mint trust root is roughly similar to the administrator roles in that contract.
+
+Our design does not have a `pauser` or a `blacklister` role. The blacklister role doesn't appear to be practical given that our chain is a privacy chain. We did not determine that the `pauser` role would serve a useful purpose in our system.
+## Set of Signers vs Threshold Signatures
+
+An alternate proposal would be to use a Threshold Signature, such as [Flexible Round-Optimized Schnorr Threshold Signatures (FROST)](https://datatracker.ietf.org/doc/pdf/draft-irtf-cfrg-frost-01.pdf). The reason we do not intend to use Threshold Signatures at this time is that we consider the explicit list of signers to be a desirable feature for auditability. In a threshold signature, the signer public key is a composite of all the participants, and it is not known exactly which _M_ of the _N_ participants participated to create the signature.
```

Comment:
This is true. But as a thought experiment, is there potentially some kind of knowledge we'd find desirable to be able to prove increase the trust of a threshold signature (like M out of N who know some secret that increases trust in them)

Reply:
I think that is too abstract. Is there a specific signature scheme you have in mind? What are you trying to solve here?